src/ExtranetBundle/objects/WebModules/MTExtranetCore/MTExtranetCoreEndPoint.class.php

```diff
@@ -221,7 +221,8 @@ public function Register($sSuccessURL = null, $sFailureURL = null)
             }
 
             if (true === $bDataValid) {
-                $bDataValid = $oUser->Register();
+                $registrationResult = $oUser->Register();
```

Comment:
The return type of \TDataExtranetUser::Register() should also be fixed then.

Reply:
I fixed first this method, but later restored it again. I thing, this change/fix can produce an error in several places.

Reply:
Why? It is annotated as bool but in 99% of the cases it returns string.

Reply:
i talked with @fabianbarth-esono about the refactoring for the Method Register, but we both thinked, that should be in many place error generate. But, alright, i have the method refactored and committed.

Reply:
The initial code was ok.

I meant only the the return *type* annotation - not the implementation of the method.

Reply:
Alright, Annotation updated and initial code is again here.